### PR TITLE
fix/front matter template

### PR DESCRIPTION
**Pull request by sywhb**
- fix: download content and file attachment only if included in the template
- fix: allow adding aliases to the variables in the front matter
- feat: add more variables in the front matter setting
- feat: add a optional front matter template under advanced settings
- feat: add link to our discord server in the settings

### CODE REVIEW

**Comment by jacksonh** [comment]
Doing some local testing

File: src/settings/template.ts [hunk 1]
```diff
@@ -163,6 +164,7 @@ export const renderArticleContnet = async (
   dateSavedFormat: string,
   isSingleFile: boolean,
   frontMatterVariables: string[],
```
Comment (jacksonh) [inline src/settings/template.ts:166]: Does Obsidian handle converting these into an array for us? It looks like it works well, i tested out some scenarios where i add extra whitespace and it was trimmed properly.

File: src/util.ts [hunk 2]
```diff
@@ -161,3 +161,6 @@ export const removeFrontMatterFromContent = (content: string): string => {
 
   return content.replace(frontMatterRegex, "");
 };
+
+export const snakeToCamelCase = (str: string) =>
+  str.replace(/(_[a-z])/g, (group) => group.toUpperCase().replace("_", ""));
```
Comment (jacksonh) [inline src/util.ts:166]: Does this mean numbers and `/` would be replaced?

**Review by jacksonh — APPROVED** [review]
One question about `snakeToCamelCase`, i want to test that a bit more

File: src/util.ts (hunk 2 above)
Comment (jacksonh) [reply src/util.ts:166]: Oh nevermind, this is for variables, i thought it was for escaping label names.

File: src/settings/template.ts (hunk 1 above)
Comment (sywhb) [reply src/settings/template.ts:166]: Oh, I convert and trim those in the `onChange` callback

File: src/util.ts (hunk 2 above)
Comment (sywhb) [reply src/util.ts:166]: no, i think it only converts `_` following with a lower case character into a upper case one.
For example, `_a` -> `A`

**Pull request merged**